flux-middle/src/global_env.rs

```diff
@@ -123,14 +131,21 @@ impl<'sess, 'tcx> GlobalEnv<'sess, 'tcx> {
     }
 
     pub fn lookup_fn_sig(&self, def_id: DefId) -> Result<rty::PolySig, UnsupportedFnSig> {
-        match self.fn_sigs.borrow_mut().entry(def_id) {
+        // If it's an extern_fn, resolve it to its local fn_sig's def_id
+        let did =
+            if let hash_map::Entry::Occupied(entry) = self.extern_fns.borrow_mut().entry(def_id) {
+                *entry.get()
+            } else {
+                def_id
+            };
```

Comment:
It's an overkill to use the entry API here a simple `get` would do

```suggestion
        let did = *self.extern_fns.borrow_mut().get(&def_id).unwrap_or(&def_id);
```